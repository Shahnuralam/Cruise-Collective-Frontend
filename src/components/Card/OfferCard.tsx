import { useSession } from "next-auth/react";
import Link from "next/link";
import React from "react";

const OfferCard = ({
  offer,
  termsAndConditionsModalData,
  setTermsAndConditionsModalData,
  setOpenLoginModal,
  setCouponModalData,
}) => {
  const { id, attributes } = offer;
  const {
    nights,
    price,
    offer_price,
    expiry_date,
    title,
    departure,
    departure_date,
    destinations,
  } = attributes;

  const { data: session } = useSession();
  console.log("special offer single", offer);

  return (
    <div className="cruise-card grid grid-cols-1 md:grid-cols-3 bg-cruise-texture mb-6">
      <div
        className={`cruise-image bg-center bg-cover cruise-card-bg-img h-52 sm:h-96 md:h-auto relative`}
      >
        <div
          className="absolute top-0 px-3 md:px-7"
          style={{ background: "rgba(255, 255, 255, 0.30)" }}
        >
          <img src="/images/po-cruise-line-card.png" alt="" />
        </div>
      </div>

      {/* Details */}
      <div className="cruise-details col-span-2 w-full flex flex-col p-2 md:p-5 justify-between">
        {/* Card Header */}
        <div className="card-header bg-[#EFEDE4] p-4 relative">
          <div className="flex bg-dark-icon-offer flex-col md:flex-row">
            <div className="text-black mb-5 w-full md:w-3/5">
              <div className="grid grid-cols-2 md:grid-cols-1">
                <p className="text-black text-base md:text-2xl mb-3 font-semibold">
                  {nights} Nights
                </p>
                <div className="text-base block md:hidden uppercase apercu_medium mb-3">
                  DEPARTING: {departure_date}
                </div>
              </div>

              <div className="text-2xl md:text-[28px] mb-4">{title}</div>
              {/* <div>EXPIRES {cruiseLineItem?.expires}</div> */}
              <div className="text-base hidden md:block uppercase apercu_medium mb-3">
                DEPARTING: {departure_date}
              </div>
              <div className="text-base uppercase apercu_medium mb-3">
                DEPARTURE PORT: {departure?.data?.attributes?.title}
              </div>
              <div className="text-base uppercase apercu_medium mb-5">
                DESTINATIONS:{" "}
                {destinations?.data?.map((item, indx) => (
                  <span key={item.id}>
                    {item?.attributes?.title}{" "}
                    {indx !== destinations.data.length - 1 && (
                      <>&nbsp;.&nbsp;</>
                    )}{" "}
                  </span>
                ))}
              </div>
            </div>
            <div className="flex w-full justify-center md:w-2/5">
              <div className="">
                <div className="uppercase text-lg font-bold pl-4 apercu_medium">
                  From
                </div>
                <div className="flex justify-center md:justify-start">
                  <div
                    className={`text-xl md:text-2xl xl:text-4xl text-cruise mr-7 ${
                      offer_price ? "line-through" : ""
                    }`}
                  >
                    £{price}
                  </div>
                  <div className=" text-xl md:text-2xl xl:text-4xl">
                    {offer_price && <>£{offer_price}</>}
                  </div>
                </div>
                {/* Button Section */}
                <div className="button-section pt-10 pb-3">
                  {!session?.user?.email && (
                    <>
                      <button className="border-cruise border-[3px] text-black hover:bg-cruise hover:underline">
                        <label
                          onClick={(e) => {
                            setOpenLoginModal(true);
                            e.stopPropagation();
                          }}
                          className="cursor-pointer tex-xl xl:text-[27px] block py-1.5 px-8"
                          htmlFor="login_modal_id"
                        >
                          Login to claim
                        </label>
                      </button>
                    </>
                  )}
                  {session?.user?.email && (
                    <>
                      <Link href={`/cruise-line-card-detail/${id}`}>
                        <button className="border-[#FF9A31] border-[3px] py-2 px-7 text-black tex-xl xl:text-[27px] hover:bg-cruise hover:underline">
                          View More
                        </button>
                      </Link>
                      {/* <button
                            onClick={() => {
                              setTermsAndConditionsModalData(cruiseLineItem);
                            }}
                            className="border-[#FF9A31] border-4"
                          >
                            <label
                              htmlFor="cruise-line-item-terms-and-conditions-modal"
                              className="cursor-pointer block w-full p-2 text-black text-[27px]"
                            >
                              Terms & Conditions
                            </label>
                          </button> */}
                    </>
                  )}
                </div>
                <div className="px-2 py-1 font-semibold text-base text-center apercu_medium uppercase">
                  EXPIRES {expiry_date}
                </div>
                {/* <div className="absolute right-5">
                    <SvgCruiseImage />
                  </div> */}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default OfferCard;
