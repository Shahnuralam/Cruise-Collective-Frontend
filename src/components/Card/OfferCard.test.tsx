import React from 'react';
import renderer from 'react-test-renderer';
import OfferCard from '../../components/Card/OfferCard';
import { useSession } from 'next-auth/react';

jest.mock('next-auth/react', () => ({
    useSession: jest.fn(),
}));

describe('OfferCard Component', () => {
    test('renders correctly', () => {
        // Mock the necessary dependencies or provide any required context
        const mockUseSession = {
            data: { user: { email: 'test@example.com' } },
        };
        (useSession as jest.Mock).mockReturnValue(mockUseSession);

        const offer = {
            id: 1,
            attributes: {
                slug: 'special-offer-slug',
                nights: 7,
                price: 1000,
                offer_price: 800,
                expiry_date: '2023-12-31', // Provide a valid expiry date
                title: 'Special Offer',
                departure_date: '2023-01-01', // Provide a valid departure date
                Saving: '45% Off', // Provide a valid departure date
                departure: {
                    data: {
                        attributes: {
                            title: 'Departure Port',
                        },
                    },
                },
                destinations: {
                    data: [
                        {
                            id: 1,
                            attributes: {
                                title: 'Destination 1',
                                type: 'place',
                            },
                        },
                        {
                            id: 2,
                            attributes: {
                                title: 'Destination 2',
                                type: 'place',
                            },
                        },
                    ],
                },
            },
        };

        const component = renderer.create(
            <OfferCard
                offer={offer}
                termsAndConditionsModalData={null}
                setTermsAndConditionsModalData={() => { }}
                setOpenLoginModal={() => { }}
                source="special_offer"
            />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});