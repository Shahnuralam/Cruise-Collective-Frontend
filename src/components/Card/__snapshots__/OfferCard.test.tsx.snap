// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard Component renders correctly 1`] = `
<a
  href="/cruise-line-/special-offer-slug"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="cruise-card grid grid-cols-1 md:grid-cols-3 bg-cruise-texture mb-6"
  >
    <div
      className="cruise-image bg-center bg-cover cruise-card-bg-img h-52 sm:h-96 md:h-auto relative"
      style={
        {
          "backgroundImage": "url(undefined)",
        }
      }
    >
      <div
        className="absolute top-0 px-3 md:px-7 logo-background-color"
      >
        <img
          alt=""
          height="150"
          width="150"
        />
      </div>
    </div>
    <div
      className="cruise-details col-span-2 w-full flex flex-col p-2 md:p-5 justify-between"
    >
      <div
        className="card-header bg-[#EFEDE4] p-4 relative"
      >
        <div
          className="flex bg-dark-icon-offer flex-col md:flex-row"
        >
          <div
            className="text-black tracking-[1.54px] mb-5 w-full md:w-3/5"
          >
            <div
              className="grid grid-cols-2 md:grid-cols-1"
            >
              <p
                className="text-black !tracking-[0px] text-base md:text-2xl mb-3 font-normal"
              >
                7
                 Nights
              </p>
              <div
                className="text-sm block  md:hidden uppercase apercu_medium mb-3"
              >
                DEPARTING: 
                01.01.2023
              </div>
            </div>
            <div
              className="text-2xl !tracking-[0px]  md:text-[24px] mb-4"
            >
              Special Offer
            </div>
            <div
              className="text-sm hidden md:block uppercase apercu_medium mb-3"
            >
              DEPARTING: 
              01.01.2023
            </div>
            <div
              className="text-sm uppercase apercu_medium mb-3"
            >
              Saving: 
              45% Off
            </div>
            <div
              className="text-sm uppercase apercu_medium mb-5"
            >
              DESTINATIONS:
               
              <span>
                Destination 1
                <span
                  className="mx-1 relative -top-[4px]"
                >
                  .
                </span>
              </span>
              <span>
                Destination 2
              </span>
            </div>
          </div>
          <div
            className="flex w-full justify-center md:w-2/5"
          >
            <div
              className=""
            >
              <div
                className="uppercase text-xs font-bold pl-4 apercu_medium"
              >
                From
              </div>
              <div
                className="flex "
              >
                <div
                  className="text-xl md:text-2xl xl:text-2xl text-cruise mr-7 line-through"
                >
                  £
                  1,000
                </div>
                <div
                  className="text-xl md:text-2xl xl:text-2xl"
                >
                  £
                  800
                </div>
              </div>
              <div
                className="button-section pt-7 pb-3"
              >
                <button
                  className="border-[#FF9A31] border-b-[3px] py-2 px-7 text-black tex-xl xl:text-[27px] hover:bg-cruise "
                >
                  View More
                </button>
              </div>
              <div
                className=" px-2 py-1 font-semibold text-sm tracking-[1.54px] text-center apercu_medium uppercase"
              >
                EXPIRES 
                31.12.2023
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
